domains/client/domain-executor/src/lib.rs

```diff
@@ -101,6 +101,7 @@ mod system_gossip_message_validator;
 #[cfg(test)]
 mod tests;
 mod utils;
+pub mod xdm_validator;
```

Comment:
Does `xdm_validator` have to be put in `domain-executor`? I'd prefer it's positioned in `subspace-transaction-pool`.